homeassistant/components/geo_json_events/geo_location.py

```diff
@@ -48,44 +47,52 @@
 )
 
 
-async def async_setup_platform(
-    hass: HomeAssistant,
-    config: ConfigType,
-    async_add_entities: AddEntitiesCallback,
-    discovery_info: DiscoveryInfoType | None = None,
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Set up the GeoJSON Events platform."""
-    url: str = config[CONF_URL]
-    scan_interval: timedelta = config.get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL)
-    coordinates: tuple[float, float] = (
-        config.get(CONF_LATITUDE, hass.config.latitude),
-        config.get(CONF_LONGITUDE, hass.config.longitude),
-    )
-    radius_in_km: float = config[CONF_RADIUS]
-    # Initialize the entity manager.
-    manager = GeoJsonFeedEntityManager(
-        hass, scan_interval, coordinates, url, radius_in_km
-    )
+    manager: GeoJsonFeedEntityManager = hass.data[DOMAIN][entry.entry_id]
 
     @callback
     def async_add_geolocation(
-        feed_manager: GenericFeedManager,
+        feed_manager: GeoJsonFeedEntityManager,
         external_id: str,
     ) -> None:
         """Add geolocation entity from feed."""
         new_entity = GeoJsonLocationEvent(feed_manager, external_id)
         _LOGGER.debug("Adding geolocation %s", new_entity)
         async_add_entities([new_entity], True)
 
-    async_dispatcher_connect(hass, manager.signal_new_entity, async_add_geolocation)
-
-    await manager.async_init()
+    manager.listeners.append(
+        async_dispatcher_connect(hass, manager.signal_new_entity, async_add_geolocation)
+    )
+    # Do not wait for update here so that the setup can be completed and because an
+    # update will fetch data from the feed via HTTP and then process that data.
+    hass.async_create_task(manager.async_update())
```

Comment:
I think this task belongs to the config entry.
```suggestion
    entry.async_create_task(manager.async_update())
```

Reply:
OK